tests/test_momento.py

```diff
@@ -30,29 +34,32 @@ def test_create_cache_get_set_values_and_delete_cache(client: SimpleCacheClient,
 
 # Init
 def test_init_throws_exception_when_client_uses_negative_default_ttl(
-    auth_token: str,
+    configuration: Configuration, credential_provider: CredentialProvider
 ):
-    with pytest.raises(errors.InvalidArgumentError) as cm:
-        SimpleCacheClient(auth_token, -1)
-        assert cm.exception == "TTL Seconds must be a non-negative integer"
+    with pytest.raises(errors.InvalidArgumentError, match="TTL timedelta must be a non-negative integer"):
+        SimpleCacheClient(configuration, credential_provider, timedelta(seconds=-1))
 
 
 def test_init_throws_exception_for_non_jwt_token(default_ttl_seconds: int):
-    with pytest.raises(errors.InvalidArgumentError) as cm:
-        SimpleCacheClient("notanauthtoken", default_ttl_seconds)
-        assert cm.exception == "Invalid Auth token."
+    with pytest.raises(errors.InvalidArgumentError, match="Invalid Auth token."):
+        os.environ["BAD_AUTH_TOKEN"] = "notanauthtoken"
+        EnvMomentoTokenProvider("BAD_AUTH_TOKEN")
 
 
-def test_init_throws_exception_when_client_uses_negative_request_timeout_ms(auth_token: str, default_ttl_seconds: int):
-    with pytest.raises(errors.InvalidArgumentError) as cm:
-        SimpleCacheClient(auth_token, default_ttl_seconds, -1)
-        assert cm.exception == "Request timeout must be greater than zero."
+def test_init_throws_exception_when_client_uses_negative_request_timeout_ms(
+    configuration: Configuration, credential_provider: CredentialProvider, default_ttl_seconds: int
+):
+    with pytest.raises(errors.InvalidArgumentError, match="Request timeout must be greater than zero."):
+        configuration = configuration.with_client_timeout(-1)
+        SimpleCacheClient(configuration, credential_provider, default_ttl_seconds)
```

Comment:
Doesn't this take a timedelta, not an int?

How did this pass tests? Does an int get cast to a timedelta?

Reply:
Good catch-- this error message wasn't accurate. The tests passed because the mistyped argument to `with_client_timeout()` also triggered the validation failure. Also python doesn't enforce typehints. The fixture being passed in as `default_ttl_seconds` actually is a timedelta, so was fine to work with the tests despite the incorrect type hint in the test method signature.

I updated the message to specify that the timeout needs to be a timedelta, left this test as it is because it's still useful, and added a new test with the negative value correctly expressed as a timedelta. I also corrected the other calls to `with_client_timeout()` that were passing integers and corrected the typehints for tests using the `default_ttl_seconds` fixture from int to timedelta.

Reply:
I wonder if we can get the tests to enforce type hints?

Reply:
@schwern we're going to do that in short order; we don't run the type checker on the tests right now.

Reply:
@pgautier404 I've run into that issue in Ruby, too. I've gotten around it by defining subclasses of InvalidArgumentError and checking for specific ones.